### PR TITLE
GNUIndentBear.py: Fix max_line_length=0

### PR DESCRIPTION
**Pull request by srivama**
Add infinite length when
max_line_length is set to 0

Fixes coala/coala-bears#2669

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: bears/c_languages/GNUIndentBear.py [hunk 1]
```diff
@@ -193,7 +194,7 @@ def create_arguments(filename, file, config_file,
 
         # The limit is set to a number which equals to int max in C
```
Comment (jayvdb) [inline bears/c_languages/GNUIndentBear.py]: The change you have done is ok, but this doesnt solve the problem.

`max_line_length` is not used after here.
It needs to be set much earlier in the function, before it is used.

File: bears/c_languages/GNUIndentBear.py [hunk 2]
```diff
@@ -155,6 +156,11 @@ def create_arguments(filename, file, config_file,
             Any command line options the indent binary understands. They
             will be simply passed through.
         """
+
```
Comment (jayvdb) [inline bears/c_languages/GNUIndentBear.py]: Please remove this blank line

**Comment by jayvdb** [comment]
@srivama , your commit message uses `coala/coala-bears#2669`.  You need to use the full URL to the issue.

And you need to rebase.

File: bears/c_languages/GNUIndentBear.py [hunk 3]
```diff
@@ -155,6 +156,10 @@ def create_arguments(filename, file, config_file,
             Any command line options the indent binary understands. They
             will be simply passed through.
         """
+        # The limit is set to a number which equals to int max in C
```
Comment (ksdme) [inline bears/c_languages/GNUIndentBear.py]: Is this comment still relevant?

File: bears/c_languages/GNUIndentBear.py [hunk 4]
```diff
@@ -155,6 +156,10 @@ def create_arguments(filename, file, config_file,
             Any command line options the indent binary understands. They
             will be simply passed through.
         """
+        # The limit is set to a number which equals to the max size in Python
```
Comment (jayvdb) [inline bears/c_languages/GNUIndentBear.py]: It isnt the max size in Python.

See https://docs.python.org/3/library/sys.html#sys.maxsize

Note that `Py_ssize_t` isnt something that is exposed to the Python 3 language at all -- it is quite internal, so it is better to say this is an arbitrary high number.

Comment (srivama) [reply bears/c_languages/GNUIndentBear.py]: Okay. Changing it

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala-bears/pull/2675/commits/d309cbccb018a0a87793fc1c3aa276446d6d7bfb

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**